hack/quickstart-setup.sh

```diff
@@ -30,98 +63,423 @@ if [ -z $ISTIO_INSTALL_SAIL ]; then
+success "Kubernetes namespaces created successfully."
 
 # Install gateway api
-echo "Installing Gateway API in ${KUADRANT_CLUSTER_NAME}"
-${KUSTOMIZE_BIN} build ${KUADRANT_GATEWAY_API_KUSTOMIZATION} | kubectl apply -f -
+info "Installing Gateway API... 🌉"
+kubectl apply -k ${KUADRANT_GATEWAY_API_KUSTOMIZATION}
+success "Gateway API installed successfully."
 
 # Install istio
-echo "Installing Istio in ${KUADRANT_CLUSTER_NAME}"
+info "Installing Istio as a Gateway API provider... 🛫"
 if [ "$ISTIO_INSTALL_SAIL" = true ]; then
-  ${KUSTOMIZE_BIN} build ${KUADRANT_ISTIO_KUSTOMIZATION} | kubectl apply -f -
+  info "Installing Istio via Sail"
+  kubectl apply -k ${KUADRANT_ISTIO_KUSTOMIZATION}
   kubectl -n istio-system wait --for=condition=Available deployment istio-operator --timeout=300s
   kubectl apply -f ${KUADRANT_REPO_RAW}/config/dependencies/istio/sail/istio.yaml
 else
   # Create CRD first to prevent race condition with creating CR
-  ${KUSTOMIZE_BIN} build ${MGC_ISTIO_KUSTOMIZATION} > ${TMP_DIR}/doctmp
+  info "Generating Istio configuration... 🛠️"
+  kubectl kustomize ${MGC_ISTIO_KUSTOMIZATION} >${TMP_DIR}/doctmp
+  success "Istio configuration generated."
   ${YQ_BIN} 'select(.kind == "CustomResourceDefinition")' ${TMP_DIR}/doctmp | kubectl apply -f -
   kubectl -n istio-system wait --for=condition=established crd/istiooperators.install.istio.io --timeout=60s
   cat ${TMP_DIR}/doctmp | kubectl apply -f -
   kubectl -n istio-operator wait --for=condition=Available deployment istio-operator --timeout=300s
 fi
+success "Istio installed successfully."
 
 # Install cert-manager
-echo "Installing cert-manager in ${KUADRANT_CLUSTER_NAME}"
-${KUSTOMIZE_BIN} build ${KUADRANT_CERT_MANAGER_KUSTOMIZATION} | kubectl apply -f -
-echo "Waiting for cert-manager deployments to be ready"
+info "Installing cert-manager... 🛡️"
+kubectl apply -k ${KUADRANT_CERT_MANAGER_KUSTOMIZATION}
+info "Waiting for cert-manager deployments to be ready"
 kubectl -n cert-manager wait --for=condition=Available deployments --all --timeout=300s
+setupClusterIssuer
+success "cert-manager installed successfully."
 
 # Install metallb
-echo "Installing metallb in ${KUADRANT_CLUSTER_NAME}"
-${KUSTOMIZE_BIN} build ${KUADRANT_METALLB_KUSTOMIZATION} | kubectl apply -f -
-echo "Waiting for metallb-system deployments to be ready"
+info "Installing MetalLB... 🏗️"
+{
+  kubectl apply -k ${KUADRANT_METALLB_KUSTOMIZATION} 2>&1
+} | grep -v "Warning: .* deprecated" || true
 kubectl -n metallb-system wait --for=condition=Available deployments controller --timeout=300s
 kubectl -n metallb-system wait --for=condition=ready pod --selector=app=metallb --timeout=60s
-kubectl apply -n metallb-system -f - <<< "$(curl -s ${KUADRANT_REPO_RAW}/utils/docker-network-ipaddresspool.sh | bash -s -- kind)"
+info "Generating IP address pool for MetalLB..."
+generate_ip_address_pool "kind" | kubectl apply -n metallb-system -f -
+success "MetalLB installed and IP address pool generated successfully."
```

Comment:
What's the issue with calling `docker-network-ipaddresspool.sh` here?

Reply:
Hmm, i see, i just don't like the duplication, you change one, you need to change both and check both deployment paths (Quickstart vs local-setup.

Possibly outside the scope of the changes here, and could unintended issues with lcoal-setup,  but you could have `utils/docker-network-ipaddresspool.sh` source `hack/quickstart-setup.sh` and just call `generate_ip_address_pool`

`hack/quickstart-setup.sh` becomes the source of all common functions.

Reply:
yeah, wasn't a fan of it either. pushed a change now which is a bit better:

- sources via cURL if ran interactively, using same KUADRANT_REF
- If ran locally (say, while debugging/making local changes), sources from disk

you might take a look